streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -190,14 +190,20 @@ State setState(final State newState) {
             oldState = state;
 
             if (state == State.PENDING_SHUTDOWN && newState != State.DEAD) {
+                log.debug("Invalid transition from PENDING_SHUTDOWN to {}: " +
+                              "only DEAD state is a valid next state", newState);
```

Comment:
Would be better to add an explanation for ignorance?

Reply:
Sure. Why not. I just would avoid the term "invalid" because it might confuse users (they may thing something bad happens, but it's expected and not bad).

Reply:
Sounds good